### PR TITLE
Error ranking

### PR DESCRIPTION
**Pull request by kirti-d**
till now we have been showing errors in the same order in which they appear in logs but more often the program stops due to some fatal error which appear in the end in logs and most of the users will be willing to see that error first so we are showing errors in bottom up manner so that user don't have to scroll all up to the last page to find that error

### CODE REVIEW

File: src/main/webapp/pagination.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+const extraPageInFrontAndBack = Math.floor(noOfPages/2);
+const LOGS = "logs" ;
+const ERRORS = "errors";
+let fileLength = Number.MAX_VALUE;
+let noOfRecordsOnLastPage = recordsPerPage;
+let lastPage = Number.MAX_VALUE;
+let data = new Array();
+
+// decrement by 1 on pressing previous button
+function prevPage() {
```
Comment (nishantnahata) [inline src/main/webapp/pagination.js:14]: we can also use ES6 functions.
prevPage = () => {}

File: src/main/webapp/pagination.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
+let currentPage = 1;
+let next = true;
+let recordsPerPage = 3;
+const noOfPages = 5;
+const extraPageInFrontAndBack = Math.floor(noOfPages/2);
+const LOGS = "logs" ;
+const ERRORS = "errors";
```
Comment (nishantnahata) [inline src/main/webapp/pagination.js:7]: const above let

File: src/main/webapp/pagination.js [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+// change content of page 
+async function changePage(page) {
+    const logs = document.getElementById("logs").getAttribute("aria-selected");
+    const searchString = document.getElementById("searchBar").value;
+    let fileName = document.getElementById("fileName").value;
+    const fileType = logs == "true" ? LOGS : ERRORS ;
+
+    currentPage = page;
+    if(page == 1 && fileType == ERRORS){
+        fileLength = await getCount(fileName+"error");
```
Comment (nishantnahata) [inline src/main/webapp/pagination.js:36]: like other servlet. we can leave the filename logic to DocCountServlet itself. Send fileType in the request

File: src/main/webapp/pagination.js [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    }
+    const fetchedPage = getPageToBeFetched(); 
+    if(currentPage != 1 ) {
+        display();
+    }
+    if(fetchedPage != -1) {
+       
+        const params = new URLSearchParams();
+        const startAndSize = getPageStartAndSize(fileType, fetchedPage);
```
Comment (nishantnahata) [inline src/main/webapp/pagination.js:45]: make it json.
`const pageSpecs = getPageSpecs(fileType, fetchedPage);`
size will be pageSpecs.size and start will be pageSpecs.start

File: src/main/java/com/google/error_analyzer/data/constant/LogFields.java [hunk 5]
```diff
@@ -17,10 +17,10 @@
     public static final String LOG_TEXT = "logText";
     public static final String LOG_LINE_NUMBER = "logLineNumber";
     public static final String LOG = "Log";
-    public static final String INDEX = "index";
```
Comment (nishantnahata) [inline src/main/java/com/google/error_analyzer/data/constant/LogFields.java:20]: isnt it used anywhere else?

Comment (kirti-d) [reply src/main/java/com/google/error_analyzer/data/constant/LogFields.java:20]: nope i replaced it with fileName so not needed now

File: src/main/webapp/result.html [hunk 6]
```diff
@@ -0,0 +1,211 @@
+<!-- Copyright 2020 Google LLC
```
Comment (shudhatmajain) [inline src/main/webapp/result.html:1]: Let's send this file with JS files.
Similarly other UI related files.

Only keep server related files here.

**Pull request merged**